spring-integration-core/src/main/java/org/springframework/integration/config/HandlerMethodArgumentResolversHolder.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.config;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.messaging.handler.invocation.HandlerMethodArgumentResolver;
+
+/**
+ * A holder for the configured argument resolvers.
+ *
+ * @author Gary Russell
+ * @since 5.0
+ *
+ */
+public class HandlerMethodArgumentResolversHolder {
```

Comment:
Can't we use `HandlerMethodArgumentResolverComposite` from SF instead?
Or there might be really a clash that it is a `HandlerMethodArgumentResolver` per se?

Reply:
Ironically, that class is not spring-friendly (XML); it can't be configured as a `<bean/>`.

Reply:
Ha! I see 😄 
OK. Let's stay with our version!
Plus we will be sure that there won't be any conflicts since `HandlerMethodArgumentResolverComposite` is a `HandlerMethodArgumentResolver` as well.